File: rdkit-sys/wrapper/src/inchi.cc
#include "rust/cxx.h"
#include <GraphMol/GraphMol.h>
#include <GraphMol/inchi.h>

namespace RDKit {

rust::String mol_to_inchi(const std::shared_ptr<ROMol> &mol) {
	ExtraInchiReturnValues rv;
	return MolToInchi(*mol, rv);
}

rust::String mol_to_inchi_key(const std::shared_ptr<ROMol> &mol) {
	return MolToInchiKey(*mol);
}

} // namespace RDKit

File: rdkit-sys/src/bridge/inchi.rs
#[cxx::bridge(namespace = "RDKit")]
pub mod ffi {
    unsafe extern "C++" {
        include!("wrapper/include/ro_mol.h");
        include!("wrapper/include/inchi.h");

        pub type ROMol = crate::ro_mol_ffi::ROMol;

        pub fn mol_to_inchi(mol: &SharedPtr<ROMol>) -> String;
        pub fn mol_to_inchi_key(mol: &SharedPtr<ROMol>) -> String;
    }
}
